subhub/api/webhooks/stripe/customer.py

```diff
@@ -63,53 +65,58 @@ def run(self):
 
 class StripeCustomerSubscriptionCreated(AbstractStripeWebhookEvent):
     def run(self):
-        data = {
-            "event_id": self.payload.id,
-            "event_type": self.payload.type,
-            "customer_id": self.payload.data.object.customer,
-            "subscription_id": self.payload.data.object.id,
-            "current_period_start": self.payload.data.object.current_period_start,
-            "current_period_end": self.payload.data.object.current_period_end,
-            "canceled_at": self.payload.data.object.canceled_at,
-            "days_until_due": self.payload.data.object.days_until_due,
-            "default_payment_method": self.payload.data.object.default_payment_method,
-            "plan_id": self.payload.data.object.plan.id,
-            "plan_amount": self.payload.data.object.plan.amount,
-            "plan_currency": self.payload.data.object.plan.currency,
-            "plan_interval": self.payload.data.object.plan.interval,
-            "status": self.payload.data.object.status,
-            "trial_start": self.payload.data.object.trial_start,
-            "trial_end": self.payload.data.object.trial_end,
-            "tax_percent": self.payload.data.object.tax_percent,
-            "application_fee_percent": self.payload.data.object.application_fee_percent,
-            "user_id": self.payload.data.object.metadata.get("userid", None),  # why?
-        }
-        routes = [StaticRoutes.SALESFORCE_ROUTE, StaticRoutes.FIREFOX_ROUTE]
-        self.send_to_routes(routes, json.dumps(data))
+        user_id = self.payload.data.object.metadata.get("userid", None)
+        updated_customer = ""
+        if not user_id:
+            customer_id = self.payload.object.customer
+            updated_customer = stripe.Customer.retrieve(customer_id)
+            user_id = updated_customer.metadata.get("userid")
+        sub_status = False
+        if self.payload.data.object.status in ["active", "trialing"]:
+            sub_status = True
+        if user_id:
+            sfd = dict(
+                uid=user_id,
+                active=sub_status,
+                subscriptionId=self.payload.data.object.id,
+                productName=self.payload.data.object.plan.nickname,
+                eventId=self.payload.id,
+                event_id=self.payload.id,
+                eventCreatedAt=self.payload.created,
+                messageCreatedAt=int(time.time()),
+            )
+            routes = [StaticRoutes.FIREFOX_ROUTE]
+            self.send_to_routes(routes, json.dumps(sfd))
+        else:
+            logger.exception(f"userid is None for customer {updated_customer}")
```

Comment:
else logger exception again, this is correct here?

Reply:
Changed to raise ClientError so this will be logged.  Chances of this happening are extremely slim as the userid is required to setup an account in the first place.